### PR TITLE
add flake8-simplify SIM105 lint

### PR DESCRIPTION
**Pull request by danieleades**

### CODE REVIEW

File: sphinx/events.py [hunk 1]
```diff
@@ -80,12 +81,10 @@ def disconnect(self, listener_id: int) -> None:
...
              allowed_exceptions: Tuple[Type[Exception], ...] = ()) -> List:
         """Emit a Sphinx event."""
-        try:
-            logger.debug('[app] emitting event: %r%s', name, repr(args)[:100])
-        except Exception:
-            # not every object likes to be repr()'d (think
+        with suppress(Exception):
             # random stuff coming via autodoc)
-            pass
+            # not every object likes to be repr()'d (think
```
Comment (AA-Turner) [inline sphinx/events.py]: These two lines are in the wrong order?

A

File: sphinx/ext/autodoc/__init__.py [hunk 2]
```diff
@@ -460,11 +461,8 @@ def format_name(self) -> str:
 
     def _call_format_args(self, **kwargs: Any) -> str:
         if kwargs:
-            try:
+            with suppress(TypeError):  # avoid chaining exceptions, by putting nothing here
```
Comment (AA-Turner) [inline sphinx/ext/autodoc/__init__.py]:
```suggestion
            with suppress(TypeError):  # avoid chaining exceptions
```

File: sphinx/ext/inheritance_diagram.py [hunk 3]
```diff
@@ -185,13 +186,12 @@ def recurse(cls: Any) -> None:
 
             # Use first line of docstring as tooltip, if available
             tooltip = None
-            try:
+            with suppress(AttributeError):
+                # might raise AttributeError for strange classes
                 if cls.__doc__:
                     doc = cls.__doc__.strip().split("\n")[0]
                     if doc:
                         tooltip = '"%s"' % doc.replace('"', '\\"')
```
Comment (AA-Turner) [inline sphinx/ext/inheritance_diagram.py]:
```suggestion
            doc = getattr(cls, '__doc__', '').strip().split("\n")[0].replace('"', '\\"')
            if doc:
                tooltip = f'"{doc}"'
```

**Review by AA-Turner — CHANGES_REQUESTED** [review]
Looks OK barring some notes:

**Comment by AA-Turner** [comment]
@danieleades please avoid force pushing after review comments, limitations of GitHub's UI make it challenging to check what has changed compared to previous reviews. I always squash merge anyway so no issues on linear commit history.

A

File: sphinx/ext/inheritance_diagram.py (hunk 3 above)
Comment (AA-Turner) [reply sphinx/ext/inheritance_diagram.py]: This was marked as resolved but it stil seems to be the old version.

Comment (danieleades) [reply sphinx/ext/inheritance_diagram.py]: whoops! should be resolved now

**Comment by danieleades** [comment]
rebased on master

File: sphinx/ext/inheritance_diagram.py (hunk 3 above)
Comment (danieleades) [reply sphinx/ext/inheritance_diagram.py]: doesn't appear to work-

```
    def recurse(cls: Any) -> None:
        if not show_builtins and cls in py_builtins:
            return
        if not private_bases and cls.__name__.startswith('_'):
            return
    
        nodename = self.class_name(cls, parts, aliases)
        fullname = self.class_name(cls, 0, aliases)
    
        # Use first line of docstring as tooltip, if available
        tooltip = None
>       doc = getattr(cls, '__doc__', '').strip().split("\n")[0].replace('"', '\\"')
E       AttributeError: 'NoneType' object has no attribute 'strip'
```

**Comment by AA-Turner** [comment]
Closing out-of-date PR (feel free to update & re-open).

A

**Pull request closed without merging**